charms/worker/k8s/charmcraft.yaml

```
@@ -54,6 +54,19 @@ bases:
       architectures: [amd64]
 config:
   options:
+    annotations:
+      type: string
+      default: "{}"
+      description: |
+        Annotations is a JSON dictionary (key/value pairs) object that can be
+        used to add arbitrary metadata configuration to the Canonical
+        Kubernetes cluster. For more information, see the upstream Canonical
+        Kubernetes documentation about annotations:
+
+        https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/reference/annotations/
+
+        The value for this config must be a JSON object, like this:
+          e.g.: {"key1": "value1", "key2": "value2"}
```

are the annotations ever `key=value` pairs where key and value are non-strings?  
do the values for `key` or `value` ever contain the `=` character?  

If the answer to both of these is no, may i recommend:

```suggestion
      default: ""
      description: |
        Annotations is a space separated list of (key/value) pairs)  that can be
        used to add arbitrary metadata configuration to the Canonical
        Kubernetes cluster. For more information, see the upstream Canonical
        Kubernetes documentation about annotations:

        https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/reference/annotations/

        The value for this config must be a JSON object, like this:
          e.g.: key1=value1 key2=value2
```

---

Also, can you say what resources we're specifically annotating?  Are they only the worker nodes?  This PR only has changed the k8s-worker charms' charmcraft.yaml?

---

OH OH OH... these are microcluster annotations.  I understand now that these are shared across the whole cluster using microcluster